ba-80 Replace search header in inventory with search element

lyio opened the pull request:
- Include socobo-element-search
- Change demo page to use new component database according to #ba-75
- Implement some rudimentary WCT tests
- Use socobo-element-search in inventory-overview
- Implement item filtering according to the search
- Implement check all and uncheck all

lyio committed:
ba-80 Replace search header in inventory with search element
- Include socobo-element-search
- Change demo page to use new component database according to #ba-75
- Implement some rudimentary WCT tests
- Use socobo-element-search in inventory-overview
- Implement item filtering according to the search
- Implement check all and uncheck all

flashback2k14 commented:
- layout still broken
- inventory items doesn't load
- unhandled exception
  ![bug](https://cloud.githubusercontent.com/assets/6609359/11974836/4081ccb4-a968-11e5-8253-8aef4970f1cf.PNG)

lyio commented:
- As you can read, this PR has nothing to do with layout.
- The exception is caused because there are no items available
- Investigating why no items are shown, as the component by itself loads the items fine.

lyio commented:
Looks like this for me...

![socobo-inventory](https://cloud.githubusercontent.com/assets/3695928/11976591/eb403bde-a979-11e5-98d6-f17163c8588a.PNG)

Once I get bower to install everything the search header should be there as well.

lyio commented:
![socobo-inventory-filtered](https://cloud.githubusercontent.com/assets/3695928/11976699/534abe9c-a97b-11e5-9dbf-f5c7caa3d47a.PNG)

Working for me. 

Current project master with this branch as socobo-inventory.

lyio committed:
ba-80 Replace search header in inventory with search element
- Add null check when no items are present on page

lyio committed:
ba-81 Behavior of paper-dropdown changed
- Included style changes from PR 16 (https://github.com/lyio/socobo-inventory/pull/16) to fix the styling of the create dialog.
The dropdowns are working  as intended in socobo-project. When running polyserve they still behave strangely.
- Also changed labels according to ba-66
- Included style documentation block provided by sebastian

flashback2k14 commented on socobo-inventory-overview.html:
```diff
@@ -42,20 +42,23 @@
+          id="searchHeader" 
+          on-check-all-items="_handleCheckAll"
+          on-uncheck-all-items="_handleUncheckAll"
+          on-changed-search-query="_searchQueryChanged">
```
you have not implemented this event 

``` html
/** 
 * This Event is fired if the search query is cleared. <br/>
 * @event clear-search-query
 */
```

this fires after press the x button.

lyio replied:
You said, there was no such event

flashback2k14 replied:
me? can't remember. but if you look into my documentation. you see all possible things!

flashback2k14 commented:
after using master branch for the project. i see inventory items, but i still get this error: 

![bug2](https://cloud.githubusercontent.com/assets/6609359/11981325/0046b922-a9a2-11e5-9481-37f953dace72.PNG)

flashback2k14 commented:
itemList is a NodeList and a NodeList is not a real array. A simple for loop should fix that bug.

lyio committed:
ba-80 Replace search header in inventory with search element

Implement search-query-cleared event
Replace forEach iteration over list elements with for..of since Document.querySelectorAll does return a NodeList that does not implement Array.prototype methods.

lyio committed:
ba-81 Behavior of paper-dropdown changed

Changed German labels to English

lyio committed:
Merge pull request #20 from lyio/ba-81-Behavior-of-paper-dropdown-changed

ba-81 Behavior of paper-dropdown changed
- Included style changes from PR 16 (#16) to fix the styling of the create dialog. The dropdowns are working as intended in socobo-project. When running polyserve they still behave strangely.
- Also changed labels according to ba-66
- Included style documentation block provided by sebastian
ba-81 Fixed

lyio committed:
ba-80 Replace search header in inventory with search element

Add missing `paper-menu` as html import as suggested in ba-81. This
fixes the problems encountered with the dropdowns in the creation
dialog.
